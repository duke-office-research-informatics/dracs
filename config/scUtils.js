import React from "react";
import { ThemeProvider } from "styled-components";
import { mount, shallow } from "enzyme";
import { theme as defaultTheme } from "../lib/dracs.es.js";

export const shallowWithTheme = (children, customTheme) => {
  const theme = customTheme || defaultTheme;
  return shallow(children, { context: theme });
};

/* eslint-disable */
export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}

//FOR ACCESSING PORTALS VIA ENZYME -- import portal from dracs
// https://github.com/airbnb/enzyme/issues/1202#issuecomment-351639783
// function mountPortal(el){
//   const wrapper =  mountWithTheme(el);
//   wrapper.setState({ visible: true});
//   const portalInstance = wrapper.find(Portal).instance();
//   return mountWithTheme(portalInstance.props.children);
// }

//FOR ACCESSING EVENT LISTENERS ATTACHED TO DOCUMENT OR DOCUMENT.BODY
//https://github.com/airbnb/enzyme/issues/426#issuecomment-296304015
// const map = {};
// document.body.addEventListener = jest.fn((event, cb) => {
//   map[event] = cb;
// });
