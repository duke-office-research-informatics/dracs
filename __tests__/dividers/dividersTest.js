import React from "react";
import { mountWithTheme } from "../../config/scUtils.js";
import { axe } from "../../config/axe-helper.js";
import { SolidDivider, DashedDivider } from "../../lib/dracs.es.js";

describe("Dividers", () => {
  describe("SolidDivider", () => {
    it("matches snapshot", () => {
      const wrapper = mountWithTheme(<SolidDivider />);
      expect(wrapper).toMatchSnapshot();
    });

    it("returns no a11y violations", async () => {
      const wrapper = mountWithTheme(<SolidDivider />);
      const results = await axe(wrapper.html());
      expect(results).toHaveNoViolations();
    });
  });

  describe("DashedDivider", () => {
    it("matches snapshot", () => {
      const wrapper = mountWithTheme(<DashedDivider />);
      expect(wrapper).toMatchSnapshot();
    });

    it("returns no a11y violations", async () => {
      const wrapper = mountWithTheme(<DashedDivider />);
      const results = await axe(wrapper.html());
      expect(results).toHaveNoViolations();
    });
  });
});
