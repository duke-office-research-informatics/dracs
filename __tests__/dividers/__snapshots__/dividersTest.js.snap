// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dividers DashedDivider matches snapshot 1`] = `
.c0 {
  width: 100%;
  height: 0px;
  border-top: 1px dashed #e8e5e2;
}

<div
  className="c0"
/>
`;

exports[`Dividers SolidDivider matches snapshot 1`] = `
.c0 {
  width: 100%;
  height: 0px;
  border-top: 1px solid #e8e5e2;
}

<div
  className="c0"
/>
`;
